# HybridAssembly

![bash](https://img.shields.io/badge/language-bash-green)

All bioinformatic analysis was conducted on the New Zealand eScience Infrastructure [NeSI](https://github.com/nesi). FastQC used for QC of the Illumina sequencing of the 11 *S. aureus* samples.

## Illumina Assembly 
[FastQC](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/) was used to check for QC the Illumina data of all the samples checking for adaptor content and sequence quality

```#!/bin/bash -e
#SBATCH --cpus-per-task=8 --mem 50Gb --time 1:00:00 -J FASTQC_EV

module load FastQC/0.11.9

FASTQC = /pathtorawreads
OUTPUT_DIR = /processedreadsdirectory/FASTQC/

mkdir -p $OUTPUT_DIR

for file in $FASTQC/*.fq.gz;
do
if [-f $FILE ];
then echo "Running FastQC on $FILE"
fastqc $FILE -o $OUTPUTDIR
else
echo "No FASTQ files found in $FASTQ"
fi
done
echo "FastQC analysis completed for all samples"
```
[SKESA](https://github.com/ncbi/SKESA) Assembly 
Genome Assembly was completed using a [SKESA script](https://github.com/emv6/Hybrid_Assembly/blob/main/skesa.sh)

[Pilon](https://github.com/broadinstitute/pilon) was used to polish the Illumina assembly using this detailed [script](https://github.com/emv6/Hybrid_Assembly/blob/main/pilon.sh) 

[CheckM](https://github.com/Ecogenomics/CheckM), [Quast](https://github.com/ablab/quast), [BUSCO](https://busco.ezlab.org/) were used to assess the Illumina assembly quality. The detailed [script](https://github.com/emv6/Hybrid_Assembly/blob/main/CheckM_Quast_BUSCO.sh) is listed here

## Nanopore Assembly 
The Nanopore assembly was followed using this [script](https://github.com/emv6/Hybrid_Assembly/blob/main/Nanopore_Assembly.sh)\
[Guppy](https://community.nanoporetech.com/docs/prepare/library_prep_protocols/Guppy-protocol/v/gpb_2003_v1_revax_14dec2018/guppy-software-overview) - Basecaller with the Super Accurate Model \
[Filtlong](https://github.com/rrwick/Filtlong) -Remove reads shorter than 1kbp and exclude the worst 5% of the reads. \
[Porechop](https://github.com/rrwick/Porechop) - Remove any adaptors \
[NanoStat](https://github.com/wdecoster/nanostat) - Quality checking each nanopore genome \
[Chopper](https://github.com/wdecoster/chopper) - Keep only the mean read quality reads \
[Flye](https://github.com/mikolmogorov/Flye) - *de novo* assembler \
[Medaka](https://github.com/nanoporetech/medaka) - Polishing tool for each genome\
[CheckM](https://github.com/Ecogenomics/CheckM), [Quast](https://github.com/ablab/quast), [BUSCO](https://busco.ezlab.org/). The detailed [script](https://github.com/emv6/Hybrid_Assembly/blob/main/CheckM_Quast_BUSCO.sh) - Assess the Nanopore assembly quality

## Hybrid Assembly method of 11 *S. aureus* isolates 
The Hybrid assembly process was generated by using Nanopore long reads and polishing the final assembly using pilon. The following tools the full script is detailed [here](https://github.com/emv6/Hybrid_Assembly/blob/main/Hybrid_Genome_Assembly.sh)

[Guppy](https://community.nanoporetech.com/docs/prepare/library_prep_protocols/Guppy-protocol/v/gpb_2003_v1_revax_14dec2018/guppy-software-overview) - Basecalling Super Accurate Model \
[Filtlong](https://github.com/rrwick/Filtlong) - Remove reads shorter than 1kbp and exclude the worst 5% of the reads. \
[Porechop](https://github.com/rrwick/Porechop) - Removing adaptors\
[NanoStat](https://github.com/wdecoster/nanostat) - Quality Checks on Genome \
[Chopper](https://github.com/wdecoster/chopper) - Keeping on the reads defined by the mean read quality by NanoStat \
[Flye](https://github.com/mikolmogorov/Flye) - *de novo* assembler \
[Medaka](https://github.com/nanoporetech/medaka) - Polishing tool for nanopore reads\
[BWA](https://github.com/lh3/bwa) and [SAMtools](https://samtools.sourceforge.net/) - Iteratively map Illumina reads to the nanopore assembly \
[Pilon](https://github.com/broadinstitute/pilon) - Polish the final Hybrid Assembly. \
[CheckM](https://github.com/Ecogenomics/CheckM), [Quast](https://github.com/ablab/quast), [BUSCO](https://busco.ezlab.org/) were used to assess the Hybrid assembly quality using this [script](https://github.com/emv6/Hybrid_Assembly/blob/main/CheckM_Quast_BUSCO.sh)  

## Defining Spa Types
Spa Types were determined using [SpaTyper](https://github.com/mjsull/spa_typing). The list of genomes is made from the Illumina asssembled contigs.

```
sed -i '1 i\spaTyper  -f ' list_of_genomes.txt
echo "--output spa.xls" &>> list_of_genomes.txt
tr '\n' ' ' < list_of_genomes.txt > spa_command.txt
chmod +x spa_command.txt
./ spa_command.txt
```
## Genome Coverage of Illumina and Nanopore Assemblies 
To determine the genome coverage of each genome's Illumina and Nanopore assembly a [script](https://github.com/emv6/Hybrid_Assembly/blob/main/Bam_Files_Illumina_Nanopore.sh) was generated to map the illumina reads to the illumina assembly using BWA mem. SAMtools depth was used to determine genome coverage. The nanopore reads were mapped to the nanopore assembly using minimap. SAMtools depth was used to determine genome coverage. 
## [Kraken2](https://github.com/DerrickWood/kraken2/blob/master/docs/MANUAL.markdown) Assembly Analysis 
```
# Counting *S. aureus* reads into the kraken2 output files 
ls 20*/*krak | xargs -I {} grep "Staphylococcus aureus" {} | cut -f2,6 > staph_a_reads.txt

# Seperating our count file into subspecies
awk -F '\t' -v OFS=, '{if(strain[$2]){strain[$2]+=$1}else{strain[$2]=$1}}END{for (name in strain){ print name, strain[name]}}' staph_a_reads.txt
```





